### PR TITLE
LRAC-6760 Accidental page views are being sent on portlet configurations

### PR DESCRIPTION
**Pull request by fcodias**

### CODE REVIEW

File: modules/apps/analytics/analytics-client-js/src/plugins/dxp.js [hunk 1]
```diff
@@ -54,7 +59,41 @@ function dxp(analytics) {
...
+	function isConfigurationPortlet() {
+		const searchParams = getSearchParams();
+
+		const portletId = searchParams.get(PARAM_PORTLET_ID_KEY);
+
+		if (portletId === PARAM_CONFIGURATION_PORTLET_NAME) {
+			return true;
+		}
+
+		return false;
```
Comment (kienD) [inline modules/apps/analytics/analytics-client-js/src/plugins/dxp.js]: Can we just return `portletId === PARAM_CONFIGURATION_PORTLET_NAME`?

File: modules/apps/analytics/analytics-client-js/src/plugins/dxp.js [hunk 2]
```diff
@@ -54,7 +59,41 @@ function dxp(analytics) {
...
+	function isViewMode() {
+		const searchParams = getSearchParams();
+
+		const mode = searchParams.get(PARAM_MODE_KEY) || PARAM_VIEW_MODE;
+
+		if (mode === PARAM_VIEW_MODE) {
+			return true;
+		}
+
+		return false;
```
Comment (kienD) [inline modules/apps/analytics/analytics-client-js/src/plugins/dxp.js]: Similar situation to the comment above.

**Review by kienD — CHANGES_REQUESTED** [review]
Just a couple nitpicks.

File: modules/apps/analytics/analytics-client-js/src/plugins/dxp.js (hunk 1 above)
Comment (andreldm) [reply modules/apps/analytics/analytics-client-js/src/plugins/dxp.js]: This is a pattern we have to comply in Java code, if you guys don't, awesome :wink:

File: modules/apps/analytics/analytics-client-js/src/plugins/dxp.js [hunk 3]
```diff
@@ -54,7 +59,41 @@ function dxp(analytics) {
...
+	}
+
 	if (window.Liferay && window.Liferay.SPA) {
+		if (isConfigurationPortlet() || !isViewMode()) {
```
Comment (andreldm) [inline modules/apps/analytics/analytics-client-js/src/plugins/dxp.js]: `getSearchParams()` is called twice, but I guess this shouldn't be a noticeable performance hit.

**Review by andreldm — APPROVED** [review]
LGTM, I tested with DXP from master (built one or two months ago) and with both SPA enabled and disable these changes behaved as expected, good job @fcodias!

**Comment by fcodias** [comment]
Hey @andreldm and @kienD thank you for the review! Just updated with the changes and also made a improvement to check even if SPA is disabled.

Do you guys mind in take a look on it again? Thanks!

**Review by kienD — APPROVED** [review]
LGTM

**Comment by fcodias** [comment]
ci:test:sf

**Comment by fcodias** [comment]
ci:forward

**Comment by fcodias** [comment]
ci:test:relevant

**Comment by fcodias** [comment]
ci:forward

**Pull request closed without merging**